client/my-sites/people/delete-user/with-delete-user.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import page from 'page';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import useDeleteUserMutation from 'calypso/data/users/use-delete-user-mutation';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+
+const useSuccessNotice = ( isSuccess, user, isMultisite, siteSlug ) => {
+	const showNotice = React.useRef();
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	React.useEffect( () => {
+		showNotice.current = () => {
+			dispatch(
+				successNotice(
+					isMultisite
+						? translate( 'Successfully removed @%(user)s', {
+								args: { user: user.login },
+								context: 'Success message after a user has been modified.',
+						  } )
+						: translate( 'Successfully deleted @%(user)s', {
+								args: { user: user.login },
+								context: 'Success message after a user has been modified.',
+						  } ),
+					{ id: 'delete-user-notice', displayOnNextPage: true }
+				)
+			);
+			page.redirect( `/people/team${ siteSlug ? `/${ siteSlug }` : '' }` );
+		};
+	} );
```

Comment:
There should be dependencies declared here.

Reply:
Good catch 👍🏻  fixed in https://github.com/Automattic/wp-calypso/pull/51841/commits/a226504ea16ac8395c00008fb2e90a6221fb6b38